### DISCUSSION TITLE
How to set client_middleware?

### DISCUSSION THREAD

**Discussion by Yurokle** [sidekiq/sidekiq#6492] [General]
Until Sidekiq 7, we could set client middleware like so:
```
Sidekiq.client_middleware {|chain| chain.add(MyMiddleware) }
```
This piece of code would run and be applied for both sidekiq and client processes

With Sidekiq 7 changes, since we don't seem to have a direct access to the middleware chains, it looks like in order to add the same client middleware to both client (e.g. puma) and server sidekiq processes, we would now need to duplicate this code:
```
::Sidekiq.configure_client do |config|
  config.client_middleware do |chain|
    chain.add(MyMiddleware)
  end
end

::Sidekiq.configure_server do |config|
  config.client_middleware do |chain|
    chain.add(MyMiddleware)
  end
end
```
we could, of course, use something like this:
```
Sidekiq.default_configuration.client_middleware {|chain| chain.add(MyMiddleware) }
```
but that looks hacky.

Is the second piece of code the way to go? Is the new, "idiomatically correct" way to go is to duplicate the code or am I missing something? Feels a bit redundant, since unless you're not enqueueing any jobs from Sidekiq process, you would probably need to add every client middleware to the chain twice.

**Comment by mperham** [comment]
> Sidekiq.default_configuration.client_middleware {|chain| chain.add(MyMiddleware) }

This.
